README.md
# react-leaflet-bing [![npm version](https://img.shields.io/npm/v/react-leaflet-bing-v2.svg)](https://www.npmjs.com/package/react-leaflet-bing-v2)![npm downloads](https://img.shields.io/npm/dw/react-leaflet-bing-v2)
Bing layer as React component for Leaflet  build on top of [React-Leaflet](https://github.com/PaulLeCam/react-leaflet).

![Bing example](images/example.gif)

## Complete example with react-leaflet-geosearch

To get started, to be able to run the example you have execute:
```bash
npm install
```

```jsx
import React from 'react';
import { MapContainer, TileLayer, LayersControl } from 'react-leaflet'
import {BingLayer} from '../src'

const { BaseLayer, Overlay } = LayersControl;


export default class BingExample extends React.Component {

  constructor() {
    super();
    this.state = {
      geodata: null,
      isadded: false
    }
  }

  render() {
    const bing_key = "YOUR-BING-KEY"
    return (
      
        <MapContainer center={[42.09618442380296, -71.5045166015625]} zoom={2} zoomControl={true}>
          <LayersControl position='topright'>
            <BaseLayer  name='OpenStreetMap.Mapnik'>
              <TileLayer  url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"/>
            </BaseLayer>
            <BaseLayer checked name='Bing Maps Roads'>
              <BingLayer  bingkey={bing_key} type="Road"/>
            </BaseLayer>
            <BaseLayer checked name='Bing Maps Satelite'>
              <BingLayer  bingkey={bing_key} />
            </BaseLayer>
            <BaseLayer checked name='Bing Maps Satelite with Labels'>
              <BingLayer  bingkey={bing_key} type="AerialWithLabels" />
            </BaseLayer>
          </LayersControl>
        </MapContainer>
    )
  }
}
```

You can find the following example in the folder ```example```. Run the above code by executing the following scripts in package.json, with the order they are stated underneath:

1) ```build```

1) ```example```

# Options

option     | type   |   | default
-----------|--------|---|--------------
subdomains | array  |   | [0, 1, 2, 3]
type       | string |One of the following values: <ul><li> Aerial– Aerial imagery.</li><li>AerialWithLabels –Aerial imagery with a road overlay.</li><li>AerialWithLabelsOnDemand - Aerial imagery with on-demand road overlay.</li><li>CanvasDark - A dark version of the road maps.</li><li>CanvasLight - A lighter version of the road maps which also has some of the details such as hill shading disabled.</li><li>CanvasGray - A grayscale version of the road maps.</li><li>Road – Roads without additional imagery.</li></ul> see: https://msdn.microsoft.com/en-us/library/ff701724.aspx| 'Aerial'
attribution | string |   | 'Bing'
culture     | string | for supported culture codes see: https://msdn.microsoft.com/en-us/library/hh441729.aspx | ''
style       | string | REST style parameters see: https://msdn.microsoft.com/en-us/library/mt823636.aspx#Related%20Resources | ''

For more details on how to use this plugin check the example.



CHANGELOG.md
### Changelog

All notable changes to this project will be documented in this file. Dates are displayed in UTC.

Generated by [`auto-changelog`](https://github.com/CookPete/auto-changelog).

#### [v5.0.1](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v5.0.0...v5.0.1)

> 6 November 2019

- update readme [`9651d5d`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/9651d5dee73b5a7b94ce88bef215afa70280c9aa)

### [v5.0.0](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v4.2.1...v5.0.0)

> 6 November 2019

- update repo name [`b0b17a3`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/b0b17a3bc71459ecb1894e38ce92e3227cb43b5a)
- Update package.json [`fcb5f98`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/fcb5f98bcb753737d67b00e7f1f47bc246137028)

#### [v4.2.1](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v4.1.0...v4.2.1)

> 6 November 2019

- release version 4.2.1 [`8e2a335`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/8e2a3357333a7197541609f8fb5ab3e9131833f7)
- update webpack and npm packages [`94d39b9`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/94d39b9090cbf2fa2adb8deedad6106771c44e6f)
- fix repo for moving to T.A. Geoforce npm [`2a4f2d9`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/2a4f2d90e439824a2072dab07d80ad612993cd00)

#### [v4.1.0](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v4.0.0...v4.1.0)

> 14 October 2018

- Set culture when requesting metadata [`52e29c4`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/52e29c42debba3051040b5b50168f022b48b8d8f)
- Forgot to build [`ab00ea1`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/ab00ea17824c902bccee73f7dce2728454587496)
- Add options to readme [`1666673`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/16666735de0b8513e6535494bb5a83a05c284687)

### [v4.0.0](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v3.1.0...v4.0.0)

> 14 October 2018

- added support for react-leaflet@2 [`#6`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/pull/6)
- update to packages [`f3a6035`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/f3a6035768d911fcaddbdfd0024290500bb6458c)

#### [v3.1.0](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v3.0.0...v3.1.0)

> 26 May 2018

- update to version 3.1.0 [`ac84df1`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/ac84df1ee593e09185c8314fff6ba9f64d77cb9f)

### [v3.0.0](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v2.0.0...v3.0.0)

> 12 September 2017

- made PropTypes to be imported from prop-types package [`#3`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/pull/3)
- update leaflet [`398aedf`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/398aedf5c24fd68405fccbae90fbecb5109ef5e4)
- added gif example [`553014b`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/553014b866a261c2e857778bd81b5ab2975761e4)
- added npm version img [`6884fe8`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/6884fe82747e56413cb203bc10d2b9f8267335f5)

### [v2.0.0](https://github.com/TA-Geoforce/react-leaflet-bing-v2/compare/v1.1.0...v2.0.0)

> 29 December 2016

- updated for leaflet v.1.0.2 [`530d3e1`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/530d3e17e56c2ba36bb2d79ad8efee12768bc8fe)
- added npm publish [`7e536e6`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/7e536e64b1cef84a4a74dc7b72cd3a1870f63a8c)

#### v1.1.0

> 27 July 2016

- First commit [`2bb961a`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/2bb961a69cc9e7f5119d6d9e19a668c06c4e3eaa)
- cleaned [`51a3e68`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/51a3e680965e3c754410d5ff3efdd979fd092b38)
- Initial commit [`70212e9`](https://github.com/TA-Geoforce/react-leaflet-bing-v2/commit/70212e99d2d437b778fb54174248408604291a3d)


__tests__/Bing.test.js
import React from 'react';
import { MapContainer, LayersControl } from "react-leaflet";
import { BingLayer } from '../src'
import { mount } from "enzyme"
import Enzyme from "enzyme"
import Adapter from "@wojtekmaj/enzyme-adapter-react-17"

Enzyme.configure({ adapter: new Adapter() })

const { BaseLayer } = LayersControl;

describe("Check if Bing renders", () => {
    it("Simple render of component", () => {
        const wrapper = mount(
            <MapContainer center={[42.09618442380296, -71.5045166015625]} zoom={2} zoomControl={true}>
                <LayersControl position='topright'>
                    <BaseLayer checked name='Bing Maps Roads'>
                        <BingLayer bingkey="your-key" type="Road"/>
                    </BaseLayer>
                 </LayersControl>
            </MapContainer>
        )
        expect((wrapper).find(BingLayer).prop('bingkey')).toEqual("your-key");
        expect((wrapper).find(BingLayer).prop('type')).toEqual("Road");
    })
})

example/bing.js
import React from 'react';
import { MapContainer, TileLayer, LayersControl } from 'react-leaflet'
import {BingLayer} from '../src'

const { BaseLayer } = LayersControl;


export default class BingExample extends React.Component {

  constructor() {
    super();
    this.state = {
      geodata: null,
      isadded: false
    }
  }

  render() {
    const bing_key = "YOUR-BING-KEY"
    return (
        <MapContainer center={[42.09618442380296, -71.5045166015625]} zoom={2} zoomControl={true}>
          <LayersControl position='topright'>
            <BaseLayer name='OpenStreetMap.Mapnik'>
              <TileLayer url="http://{s}.tile.osm.org/{z}/{x}/{y}.png"/>
            </BaseLayer>
            <BaseLayer checked name='Bing Maps Roads'>
              <BingLayer bingkey={bing_key} type="Road"/>
            </BaseLayer>
            <BaseLayer checked name='Bing Maps Satelite'>
              <BingLayer bingkey={bing_key} />
            </BaseLayer>
            <BaseLayer checked name='Bing Maps Satelite with Labels'>
              <BingLayer bingkey={bing_key} type="AerialWithLabels" />
            </BaseLayer>
          </LayersControl>
        </MapContainer>
    )
  }
}
